File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string ans = "";
        for(string s:strs) {
            ans += to_string(s.length());
            ans += "#";
            ans += s;
        }
        return ans;
    }

    vector<string> decode(string s) {
        int i=0, n = s.length();
        vector<string>ans;
        while(i<n) {
            int j = i;
            while(s[j] != '#') {
                j++;
            }
            int len = stoi(s.substr(i,j));
            ans.push_back(s.substr(j+1, len));
            i = j+1+len;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sqrtx/submission-2.java
class Solution {
    public int mySqrt(int x) {
        long s=0,e=x,mid, ans=1;
        while(s<=e) {
            mid = (s+e)/2;
            if(mid*mid<=x) {
                s = mid+1; 
                ans = mid;
            } else {
                e = mid - 1;
            }
        }
        return (int)ans;
    }
}